extensions/amp-fit-text/0.2/amp-fit-text.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FitText} from './fit-text';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+
+import {dict} from '../../../src/utils/object';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-fit-text';
+
+class AmpFitText extends PreactBaseElement {
+  /** @override */
+  init() {
+    const attributeOb = new MutationObserver(() => {
+      this.mutateProps(
+        dict({
+          'maxFontSize': getLengthNumeral(
+            this.element.getAttribute('max-font-size')
+          ),
+          'minFontSize': getLengthNumeral(
+            this.element.getAttribute('min-font-size')
+          ),
+        })
+      );
+    });
+    attributeOb.observe(this.element, {
+      attributeFilter: ['min-font-size', 'max-font-size'],
```

Comment:
This should require `attributes: true`, option, too.

Reply:
I'm not sure this is needed since `attributes: true` is implied and restricted to the given list, no? It still works without it.

Reply:
It's listed in https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#Exceptions, but the [spec](https://dom.spec.whatwg.org/#dom-mutationobserver-observe) allows it to default. I wonder when this was changed, or it the wiki was just done incorrectly.

Reply:
I'll play it on the safe side and add it in then?